### ISSUE TITLE
Docs: [unified-signatures] remove invalid examples from docs

### ISSUE THREAD

**Issue by SuperSodaSea** [typescript-eslint/typescript-eslint#6284]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.9.4&sourceType=module&code=CYUwxgNghgTiAEAzArgOzAFwJYHtVIAoA6EqALnlWQFsAjEGAbQF0BKCgNxy2AG4BYAFChIsBCnTY8hEkXLwAzhhhZUAcxbt4XHgMFCR0OEjSZc+NcVIUqdBps7c+B8EfGmpFq0VoUlK9QdtJz0gA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6WJygM0sQBNaySgHMmAQ3yxoKdGADakRNGgB7aJAA0mKCKZrEAEW5cliFvEIA5MQFt+ABTHRbifEtQZ8cRGAC+AXXA-EF8gA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
declare function f(...a: number[]): void;
declare function f(...a: string[]): void;

declare function g(...a: number[]): void;
declare function g(...b: string[]): void;
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/unified-signatures": ["error", { "ignoreDifferentlyNamedParameters": true }]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

In [`unified-signatures`' document](https://typescript-eslint.io/rules/unified-signatures/):
```ts
// "@typescript-eslint/unified-signatures": ["error", { "ignoreDifferentlyNamedParameters": true }]

// Incorrect
function f(...a: number[]): void;
function f(...b: string[]): void;

// Correct
function f(...a: number[]): void;
function f(...a: string[]): void;
```

In my opinion the document is also wrong. The first one with different name should be correct, and the second one with the same name should be incorrect.

### Actual Result

No error is given, no matter `ignoreDifferentlyNamedParameters` is `true` or `false`.

**Comment by bradzacher** [comment]
The rule doesn't check rest parameters. If you change it to normal param it works as expected.
```ts
declare function f(a: number[]): void;
declare function f(a: string[]): void;

declare function g(a: number[]): void;
declare function g(b: string[]): void;
``` [playground](https://typescript-eslint.io/play/#ts=4.9.4&sourceType=module&code=CYUwxgNghgTiAEAzArgOzAFwJYHtVIAooAueVZAWwCMQYBtAXQEpSA3HLYAbgChRJYCFOmx5CJeAGcMMLKgDmjFvHadefcNDhI0mXPnlFS5arSVsO3DQO3C9Yw1VLTZC8ysu8gA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6WJygM0sQBNaySgHMmAQ3yxoKdGADakRNGgB7aJAA0mKCKZrEAEW5cliFvEIA5MQFt+ABTHRbifEtQZ8cRGAC+AXXA-EF8gA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA)

The docs probably need to be updated to remove that "bad" case.

This is the reason that rest param aren't checked: https://github.com/Microsoft/TypeScript/issues/5077
